### PR TITLE
feature(login): Implement Login Modal for NuxtJS App

### PR DESCRIPTION
**Pull request by kibo-geetanshuchhabra**
@kibo-kevinwatts  @kibo-sushant

### CODE REVIEW

File: components/LoginModal.vue [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            data-testid="log-in-button"
+          >
+            Log In
+          </SfButton>
+        </form>
+        <SfButton
+          class="sf-button--text action-button"
+          data-testid="forgotten-password-button"
+          @click="
+            () => {
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: define a method in setup to toggle state changes, not within the template

File: components/LoginModal.vue [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        <div class="aside">
+          <SfHeading
+            title="Don't have an account yet?"
+            :level="3"
+            class="aside__heading"
+          />
+          <SfButton
+            class="sf-button--text"
+            data-testid="register-now-button"
+            @click="isLogIn = false"
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: define a method in setup to toggle state changes, not within the template

File: components/LoginModal.vue [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            class="sf-button--full-width form__submit"
+            data-testid="create-acount-button"
+          >
+            Create an account
+          </SfButton>
+        </form>
+        <SfButton
+          class="sf-button--text action-button"
+          data-testid="log-in-account"
+          @click="isLogIn = true"
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: define a method in setup to toggle state changes, not within the template

File: components/LoginModal.vue [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+      isLogin,
+      isLoginModalOpen,
+      toggleLoginModal,
+      isForgotten,
+      closeModal,
+      setIsForgottenValue,
+      barTitle,
+    }
+  },
+  data() {
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: remove data, any data needed should be returned in the setup method

File: components/LoginModal.vue [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+          >
+            Register now
+          </SfButton>
+        </div>
+      </div>
+      <div v-else-if="isForgotten">
+        <p>{{ $t("Forgot Password") }}</p>
+        <form class="form" @submit.prevent="() => false">
+          <SfInput
+            v-model="form.username"
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue:74]: Be more consistent in this template v-model binding, some are `form.` and some are not.

**Review by kibo-kevinwatts — APPROVED** [review]

File: components/LoginModal.vue (hunk 1 above)
Comment (kibo-geetanshuchhabra) [reply components/LoginModal.vue]: Resolved

File: components/LoginModal.vue (hunk 2 above)
Comment (kibo-geetanshuchhabra) [reply components/LoginModal.vue]: Resolved

File: components/LoginModal.vue (hunk 3 above)
Comment (kibo-geetanshuchhabra) [reply components/LoginModal.vue]: Resolved

File: components/LoginModal.vue (hunk 4 above)
Comment (kibo-geetanshuchhabra) [reply components/LoginModal.vue]: Resolved

File: components/LoginModal.vue (hunk 5 above)
Comment (kibo-geetanshuchhabra) [reply components/LoginModal.vue:74]: Resolved

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**